packages/common/src/types/types.ts

```diff
@@ -0,0 +1,171 @@
+import ApolloClient, {
+  ApolloQueryResult,
+  ApolloError,
+  FetchPolicy,
+  WatchQueryFetchPolicy,
+  ErrorPolicy,
+  FetchMoreOptions,
+  FetchMoreQueryOptions,
+  PureQueryOptions,
+  MutationUpdaterFn,
+  NetworkStatus,
+  ObservableQuery
+} from 'apollo-client';
+import { DocumentNode, GraphQLError } from 'graphql';
+
+/* Common types */
+
+export type OperationVariables = Record<string, any>;
+
+export type Context = Record<string, any>;
+
+export interface ExecutionResult<T = Record<string, any>> {
+  data?: T;
+  extensions?: Record<string, any>;
+  errors?: GraphQLError[];
+}
+
+/* Query types */
+
+export interface BaseQueryOptions<TVariables = OperationVariables> {
+  ssr?: boolean;
+  variables?: TVariables;
+  fetchPolicy?: WatchQueryFetchPolicy;
+  errorPolicy?: ErrorPolicy;
+  pollInterval?: number;
+  client?: ApolloClient<any>;
+  notifyOnNetworkStatusChange?: boolean;
+  context?: Context;
+  partialRefetch?: boolean;
+  returnPartialData?: boolean;
+}
+
+export interface QueryFunctionOptions<
+  TData = any,
+  TVariables = OperationVariables
+> extends BaseQueryOptions<TVariables> {
+  displayName?: string;
+  skip?: boolean;
+  onCompleted?: (data: TData) => void;
+  onError?: (error: ApolloError) => void;
+}
+
+export type ObservableQueryFields<TData, TVariables> = Pick<
+  ObservableQuery<TData, TVariables>,
+  | 'startPolling'
+  | 'stopPolling'
+  | 'subscribeToMore'
+  | 'updateQuery'
+  | 'refetch'
+  | 'variables'
+> & {
+  fetchMore: (<K extends keyof TVariables>(
+    fetchMoreOptions: FetchMoreQueryOptions<TVariables, K> &
+      FetchMoreOptions<TData, TVariables>
+  ) => Promise<ApolloQueryResult<TData>>) &
+    (<TData2, TVariables2, K extends keyof TVariables2>(
+      fetchMoreOptions: { query?: DocumentNode } & FetchMoreQueryOptions<
```

Comment:
Wow, Prettier is doing an especially bad job of formatting these types.

Reply:
Very much 🙁